### PR TITLE
Use std::min replace min

### PR DESCRIPTION
**Pull request by KangLin**
Use std::min replace min

### CODE REVIEW

File: common/rdr/HexOutStream.cxx [hunk 1]
```diff
@@ -41,7 +40,7 @@ bool HexOutStream::flushBuffer()
 {
   while (sentUpTo != ptr) {
     uint8_t* optr = out_stream.getptr(2);
-    size_t length = min(ptr-sentUpTo, out_stream.avail()/2);
+    size_t length = std::min(ptr-sentUpTo, (int64_t)out_stream.avail()/2);
```
Comment (CendioOssman) [inline common/rdr/HexOutStream.cxx:43]: `int64_t` seems like an odd choice here as neither the left or right side as that type. Why not `size_t` or `ptrdiff_t`?

File: common/rdr/HexInStream.cxx [hunk 2]
```diff
@@ -24,11 +24,10 @@
 #include <rdr/HexInStream.h>
 #include <rdr/Exception.h>
 #include <rfb/util.h>
+#include <algorithm>
```
Comment (CendioOssman) [inline common/rdr/HexInStream.cxx:27]: We try to sort system includes before the local ones, and a blank line to separate them. Makes it easier to get an overview of the includes.

**Review by CendioOssman — CHANGES_REQUESTED** [review]
The PR looks mostly fine, but I'm unclear to the motivation. You mention "cross-platform", but how does this change improve that?

File: common/rdr/HexInStream.cxx [hunk 3]
```diff
@@ -21,13 +21,16 @@
...
 
+#include <algorithm>
 #include <rdr/HexInStream.h>
 #include <rdr/Exception.h>
 #include <rfb/util.h>
 
-using namespace rdr;
+#if defined(_MSC_VER)
+#undef min
+#endif
```
Comment (CendioOssman) [inline common/rdr/HexInStream.cxx:31]: This is a surprising new addition. Are you saying that std::min cannot be safely used in Visual Studio?

File: common/rdr/HexInStream.cxx [hunk 4]
```diff
@@ -37,12 +40,11 @@ HexInStream::HexInStream(InStream& is)
 HexInStream::~HexInStream() {
 }
 
-
 bool HexInStream::fillBuffer() {
   if (!in_stream.hasData(2))
     return false;
-
-  size_t length = min(in_stream.avail()/2, availSpace());
+
```
Comment (CendioOssman) [inline common/rdr/HexInStream.cxx:46]: Some whitespace snuck in here.

**Review by CendioOssman — CHANGES_REQUESTED** [review]

File: common/rdr/HexInStream.cxx (hunk 3 above)
Comment (KangLin) [reply common/rdr/HexInStream.cxx:31]: In fact, the mix macro is defined in the minwindef.h header file, while Windows.h references minwindef.h. The mix macro is defined as follows:
```
#ifndef NOMINMAX
 
#ifndef max
#define max(a,b)            (((a) > (b)) ? (a) : (b))
#endif
 
#ifndef min
#define min(a,b)            (((a) < (b)) ? (a) : (b))
#endif
 
#endif  /* NOMINMAX */
```

Maybe it's better to define  NOMINMAX

**Review by CendioOssman — CHANGES_REQUESTED** [review]
I think the code looks good now. But I'm still not clear on the rationale. What was the reason you started working on this fix?

**Comment by KangLin** [comment]
I am will use to msvc.

在 2024-07-29 22:06:42，"Pierre Ossman (ThinLinc team)" ***@***.***> 写道：


@CendioOssman requested changes on this pull request.

I think the code looks good now. But I'm still not clear on the rationale. What was the reason you started working on this fix?

**Review by CendioOssman — APPROVED** [review]
> - There are standard functions available, so why define them yourself?

Generally, that's a good principle. But `std::min()` has some bad behaviours based on the fact that it is implemented as a template, which means it isn't always a good fit.

We also don't want to avoid creating unnecessary history, as that makes `git blame` annoying to use. In other words, there should be a decently sized cleanup.

That said, this didn't change too many lines. So I can accept this if it makes things easier for you.

> - I am will use to msvc.

Ah. That's not one of our normally used compilers, but it's always a plus if we're more compatible.

**Comment by KangLin** [comment]
> But `std::min()` has some bad behaviours based on the fact that it is implemented as a template, which means it isn't always a good fit.

Please describe any bad behaviours？

**Pull request merged**